tests/integration.rs

```diff
@@ -1009,6 +1009,87 @@ mod tests {
         });
     }
 
+    #[cfg_attr(not(feature = "mmio"), test)]
+    fn test_cpu_topology_421() {
+        test_cpu_topology(4, 2, 1);
+    }
+
+    #[cfg_attr(not(feature = "mmio"), test)]
+    fn test_cpu_topology_142() {
+        test_cpu_topology(1, 4, 2);
+    }
+
+    #[cfg_attr(not(feature = "mmio"), test)]
+    fn test_cpu_topology_262() {
+        test_cpu_topology(2, 6, 2);
+    }
+
+    fn test_cpu_topology(threads_per_core: u8, cores_per_package: u8, packages: u8) {
```

Comment:
Just out of curiosity, why aren't you making the `dies_per_package` a variable?

Reply:
Because there is no way to query it from the guest.